service/history/historyEngine.go

```diff
@@ -152,6 +163,7 @@ func NewEngineWithShardContext(shard ShardContext, visibilityMgr persistence.Vis
 		}),
 		metricsClient:        shard.GetMetricsClient(),
 		historyEventNotifier: historyEventNotifier,
+		initiator:            sysworkflow.NewInitiator(frontendClient, shard.GetConfig().NumSysWorkflows()),
```

Comment:
It would be good to make this config truly dynamic which does not need a restart of the roles.